src/openforms/registrations/contrib/zgw_apis/plugin.py

```diff
@@ -578,7 +578,9 @@ def register_submission_to_objects_api(
         )
 
         # In a follow up PR: the group will be configurable:
-        api_group = ObjectsAPIGroupConfig.objects.order_by("pk").first()
+        api_group = options.get(
+            "objects_api_group", ObjectsAPIGroupConfig.objects.order_by("pk").first()
+        )
```

Comment:
I'd prefer to remove this default/fallback behaviour entirely so that all configuration is explicit, and we can then also define this accordingly in the options type declarations.

```py
api_group = options["objects_api_group"]
assert api_group is not None
```

Reply:
Would it also work if I add it to the if statement here (https://github.com/open-formulieren/open-forms/blob/master/src/openforms/registrations/contrib/zgw_apis/plugin.py#L443-L447), so that it will only try to register the submission if the API group is part of the options?

Reply:
yes, sort of - the type checker will not be able to narrow the type of `options["objects_api_group"]` so for that you'll still need the assert